### DISCUSSION TITLE
Ping example of libp2p not working on internet , only working on local network

### DISCUSSION THREAD

**Discussion by neerajpatidar05** [libp2p/rust-libp2p#3183] [General]
I am using Libp2p library ping example and it is working fine on local network (two terminal on same machine) but it is not working on internet (two different machine with different network).

**Comment by thomaseizinger** [comment]
Are you opening and forwarding the ports on your router correctly for at least one application to reach the other?

**Comment by neerajpatidar05** [comment]
when i ran ping example it is showing output like -      Running `target/debug/examples/ping`
Local peer id: PeerId("12D3KooWAFV65NE9QuQAr5qLgmWhdTnWDwKkgTa5qbRwUaba1kKL")
Listening on "/ip4/127.0.0.1/tcp/40427"
Listening on "/ip4/172.31.12.117/tcp/40427"

and then i tried to run this example on another machine by providing that /ip4/172.31.12.117/tcp/40427 ip address.
and this is not working

**Reply by thomaseizinger** [reply]
Are those two machines on the same local network? The `172.16.0.0/16` range is for [private networks](https://en.wikipedia.org/wiki/Reserved_IP_addresses) (i.e. LAN).

To make this example work, either both machines need to be on the same local network or you need to figure out the public IP of your router and forward the ports accordingly. If that doesn't mean anything to you, you'll need to learn what it means before you can make that work! :)
